hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/CountTraverser.java

```diff
@@ -0,0 +1,192 @@
+        if (this.dedup(source)) {
+            return QueryResults.emptyIterator();
+        }
+        Iterator<Edge> flatten = this.edgesOfVertex(source, step.direction,
+                                                    step.labels, step.degree,
+                                                    step.skipDegree);
+        return new FilterIterator<>(flatten, e -> {
+            if (this.containsTraversed) {
+                // Count intermediate vertices
+                this.count[0]++;
+            }
+            return true;
+        });
+    }
+
+    private Iterator<Edge> edgesOfVertex(Id source, Directions dir,
+                                         Map<Id, String> labels,
+                                         long degree, long skipDegree) {
+        checkSkipDegree(skipDegree, degree, NO_LIMIT);
+        long queryLimit = skipDegree > 0 ? skipDegree : degree;
+        Iterator<Edge> edges = this.edgesOfVertex(source, dir, labels,
+                                                  queryLimit);
+        return ShortestPathTraverser.skipSuperNodeIfNeeded(edges, degree,
+                                                           skipDegree);
+    }
+
+    private Iterator<Edge> edgesOfVertex(Id source, Directions dir,
+                                         Map<Id, String> labels, long limit) {
+        Id[] els = labels.keySet().toArray(new Id[labels.size()]);
+        Query query = GraphTransaction.constructEdgesQuery(source, dir, els);
+        query.capacity(Query.NO_CAPACITY);
+        if (limit != NO_LIMIT) {
+            query.limit(limit);
+        }
+        return this.graph().edges(query);
+    }
+
+    private long edgesCount(Id source, Directions dir, Map<Id, String> labels) {
+        Id[] els = labels.keySet().toArray(new Id[labels.size()]);
+        Query query = GraphTransaction.constructEdgesQuery(source, dir, els);
+        query.aggregate(Aggregate.AggregateFunc.COUNT, null);
+        query.capacity(Query.NO_CAPACITY);
+        query.limit(Query.NO_LIMIT);
+        return graph().queryNumber(query).longValue();
+    }
+
+    private void checkDedup(long dedup) {
+        checkNonNegativeOrNoLimit(dedup, "dedup");
+    }
+
+    private boolean dedup(Id vertex) {
+        if (!this.needDedup()) {
+            return false;
+        }
+
+        if (this.dedupSet.contains(vertex)) {
+            // Skip vertex already traversed
+            return true;
+        } else if (!this.reachDedup()) {
+            // Record vertex not traversed before if not reach dedup
+            this.dedupSet.add(vertex);
+        }
+        return false;
+    }
+
+    private boolean needDedup() {
+        return this.dedup != 0L;
+    }
+
+    private boolean reachDedup() {
+        return this.dedup != NO_LIMIT && this.dedupSet.size() >= this.dedup;
+    }
+
+    public static class Step {
+
+        private Directions direction;
+        private Map<Id, String> labels;
```

Comment:
add properties for filtering sortkey:
`       private Map<String, Object> properties;`